lambda/helper/python/comprehendHelper.py

```diff
@@ -0,0 +1,454 @@
+import boto3
+from botocore.exceptions import ClientError
+import json
+import threading
+from helper import S3Helper
+
+
+class ComprehendHelper:
+
+    def getNumOfPages(self,
+                      textract):
+        return textract[0]['DocumentMetadata']['Pages']
+
+    #
+    # this function iterates and extracts all textual elements, aka blocks, that were
+    # identified by Textract in each page.  The result is organized as a list of strings,
+    # each containing the concatenated text of a given page.
+    #
+    def extractTextByPages(self,
+                           textract,
+                           rawPages,
+                           numOfPages):
+
+        # current service limit of Comprehend per page
+        MAX_COMPREHEND_UTF8_PAGE_SIZE = 5000
+        
+        # the textract results json file in S3 has a peculiar structure of
+        # top-level list of block lists, each appear to have a size limit of 1000.
+        # A new one is creatd and appended when the previous blocklist reaches that limit.
+        # Not sure why it is so. A blocklist may contain blocks from different document pages.
+        for blocklist in textract:
+        
+            for block in blocklist['Blocks']:
+        
+                # PAGE block type have no text, so skip those
+                if block['BlockType'] != 'PAGE':
+                    
+                    # page numbers start at 1 in Textract for png, however
+                    # for png there are no page reference since it is a single
+                    # page, in this case all blocks have page 1
+                    if numOfPages == 1:
+                        pageNumber = 1
+                    else:
+                        pageNumber = block['Page']
+                    
+                    # skip pags above limit
+                    if pageNumber > numOfPages:
```

Comment:
Could you explain when we would fall into this condition?
Also, NIT : typo in pagEs

Reply:
the textract results is a list of blocks, each with its page number.  To enforce a maximum of pages processed I'm skipping any block that has a page number above the maximum.  I could break from the loop when I encounter a block above the page limit by supposing that all following blocks will be of higher than limit pages, but I'm not too sure the textract results enforce that ordering, hence looking at each block.